src/CoreFoundation/NativeObject.cs

```diff
@@ -78,12 +82,10 @@ protected virtual void InitializeHandle (IntPtr handle)
 			this.handle = handle;
 		}
 
-		void Throw () => throw new ObjectDisposedException (GetType ().ToString ());
-
-		internal IntPtr GetCheckedHandle ()
+		public IntPtr GetCheckedHandle ()
 		{
 			if (handle == IntPtr.Zero)
-				Throw ();
+				ObjCRuntime.ThrowHelper.ThrowObjectDisposedException (GetType ());
```

Comment:
Would it be better to pass `this` to `ThrowObjectDisposedException` instead of calling `GetType ()`? That would avoid a method call (and a virtual one), which would make the IL shorter and thus more likely to be inlined by the AOT compiler. `ThrowObjectDisposedException` could do `t?.GetType ()?.ToString ()` on the input, and the end result would be the same.

Reply:
Good idea! The (virtual) call will still be made under the same condition (if `handle == `IntPtr.Zero`), but the caller's IL will be shorter (and there will be more callers than callees).